datasafe-encryption/datasafe-encryption-impl/src/test/java/de/adorsys/datasafe/encrypiton/impl/cmsencryption/CmsEncryptionServiceImplTest.java

```diff
@@ -45,9 +44,43 @@
 
     @BeforeAll
     public static void setUp() {
+        Security.addProvider(new BouncyCastleProvider());
         keyStoreAccess = getKeyStoreAccess();
     }
 
+    @Test
+    @SneakyThrows
+    public void testCmsStreamEnvelopeEncryptAndDecryptTestWithMultipleRecepients() {
+        KeyStoreAccess keyStoreAccess1 = getKeyStoreAccess();
+        KeyStoreAccess keyStoreAccess2 = getKeyStoreAccess();
+
```

Comment:
As both keystores have the same user and password this test does not proove too much. At least both of them should have different names and passwords. Further a third uses who is not in the list should try to decrypt too.

Reply:
done